Bookish/Views/Book/Catalogue.cshtml

```diff
@@ -19,31 +27,56 @@
             <th>Number of available copies</th>
             <th>Number of searches</th>
         </tr>
-        
+        </thead>
+
+        <tbody>
         @{
+            int i = -1;
+            Book btnBook = new Book();
             if (Model.AllBooks.Count >= 1)
             {
-                foreach (var bookCopy in Model.AllBooks)
+                foreach (var eachBook in Model.AllBooks)
                 {
-       
+                    i += 1;
+
                     <tr>
-                        <td>@bookCopy.Id</td>
-                        <td>@bookCopy.Title</td>
-                        <td>@bookCopy.Author</td>
-                        <td>@bookCopy.NumOfCopies</td>
-                        <td>@bookCopy.NumOfAvailableCopies</td>
-                        <td>@bookCopy.Searches</td>
-                        <td><button class="editbtn">Edit</button></td>
-                        <td><li class="nav-item">
-                            <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Index">Edit</a>
-                        </li></td>
+                        <td>@eachBook.Id</td>
+                        <td>@eachBook.Title</td>
+                        <td>@eachBook.Author</td>
+                        <td>@eachBook.NumOfCopies</td>
+                        <td>@eachBook.NumOfAvailableCopies</td>
+                        <td>@eachBook.Searches</td>
+
+                        <form id="myBook"> <input type="hidden" id="id" value="@eachBook.Id"></input> </form>
+                        
+                        @{ btnBook = eachBook;}
+                        
+                        <td><input type="button" style="background-color:cornflowerblue" value="Edit" onclick="location.href='@Url.Action("EditBook", "Book", new {parameter1 = @eachBook.Id })'" /></td>
+                        <td><input type="button" style="background-color:indianred" value="Delete" onclick="location.href='@Url.Action("DeleteBook", "Book", new {parameter1 = @eachBook.Id })'" /></td>
+
+
+
+                        @*<td>
+                            @using (Html.BeginForm("EditBook", "Book", FormMethod.Get))
```

Comment:
I don't think this needs to be in a form - you could just have a button which uses `Url.Action` in its `onclick` eg 
```
onclick="location.href='@Url.Action("EditBook", "Book", new { bookId = eachBook.Id })'"
```

I also don't think you need `i` - you'll notice I didn't need to use it in the example I just gave